src/components/Chat/Chat.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash-es';
+import cx from 'classnames';
+
+import { uniqueId } from '../../utils';
+import { CHAT_ITEM_TYPES } from './utils';
+import ChatListItem from './ChatListItem';
+
+const Chat = ({ isPastChat, items }) => {
+  const sectionClassNames = cx([
+    'slds-chat',
+    { 'slds-chat_past': isPastChat },
+  ]);
+
+  return (
+    <section role="log" className={sectionClassNames}>
+      <ul className="slds-chat-list">
+        {items.map((message, i) => {
+          const { author: messageAuthor, messageType } = message;
+          let isConsecutive = false;
+          let isFirstMessage = false;
+
+          const { author: prevAuthor, isStart: prevIsStart } = get(items, [i - 1], {});
+          const { author: nextAuthor, messageType: nextType } = get(items, [i + 1], {});
+
+          if (!isPastChat && messageType === nextType && messageAuthor === nextAuthor) {
+            isConsecutive = true;
+          }
+
+          isFirstMessage = prevIsStart || prevAuthor !== messageAuthor;
+
+          const otherProps = {
+            isConsecutive,
+            isFirstMessage,
+            isPastChat,
+          };
+
+          return <ChatListItem key={uniqueId(i, 'chat-list-item-')} {...message} {...otherProps} />;
```

Comment:
This is the same as using an `index` and will result in a re-render on every update. What about using `timestamp+type` as identifier?

Reply:
The timestamp is not really unique... Usually it's only on a per minute base.

Reply:
Would it make sense to add an `id` to the data model then? As it is, using `uniqueId` here in render does not have the desired effect and is misleading

Reply:
Yep, there is nothing else that qualifies as a key.